packages/agreement-platformstate-writer/src/consumerServiceV1.ts

```diff
@@ -149,19 +144,28 @@ const handleFirstActivation = async (
       );
     }
   } else {
-    const agreementEntry: PlatformStatesAgreementEntry = {
-      PK: primaryKey,
-      state: agreementStateToItemState(agreement.state),
-      version: incomingVersion,
-      updatedAt: new Date().toISOString(),
-      GSIPK_consumerId_eserviceId,
-      GSISK_agreementTimestamp:
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        agreement.stamps.activation!.when.toISOString(),
-      agreementDescriptorId: agreement.descriptorId,
-    };
+    if (agreement.state === agreementState.active) {
```

Comment:
I suggest [this](https://github.com/pagopa/interop-be-monorepo/pull/1331/files?diff=split&w=1#diff-a1d28d3c5e4d27ef17b7ca245c94ab9fc38950cec736b4e7eeaf6c3f9dca6a93R263) approach here as well to avoid edge cases (e.g. upgrade of a suspended agreement)